sktime/dists_kernels/base/_base.py

```diff
@@ -467,7 +467,7 @@ def transform_diag(self, X):
         diag = np.zeros(len(X_spl))
 
         for i, X_instance in enumerate(X_spl):
-            diag[i] = self.transform(X=X_instance)
+            diag[i] = self.transform(X=X_instance)[0, 0]
```

Comment:
I don't use `dist_kernels`, so confirming: is it not a behaviour change (possibly breaking), like from 2d array to scalar?

Reply:
no, not a behaviou change, it is addressing a deprecated type coercion.

diag[i] is a scalar, so we are writing an 1 x 1 matrix to a scalar field in the original code. This coercion of matrix to scalar is deprecated, and the deprecation is addressed by writing scalar to scalar instead, with the "from" scalar being the result of coercion prior to deprecation.